float32 quantity
---
bool success